numbers/logDistribution.go

```diff
@@ -9,3 +14,28 @@ func BinomialDistLog(n int, k int, p float64) float64 {
 	expression := MultiplyLog(s, f)
 	return MultiplyLog(coefficient, expression)
 }
+
+//BinomialDistLogMap returns log(BinomialDist), where log is the natural logarithm.
+//This function is similar to BinomialDistLog but passes in a map[int][]float64, where the int key
+//refers to n and the []float64 map values are the corresponding binomial coefficients for index k.
+//Useful to not recalculate the binomial coefficient each time when binomial densities must be constantly evaluated in logSpace, like in MCMC.
+func BinomialDistLogMap(n int, k int, p float64, binomMap *map[int][]float64) float64 {
+	if _, ok := (*binomMap)[n]; !ok {
+		(*binomMap)[n] = AddBinomMapEntry(n)
```

Comment:
I was wondering if we should have checks to make sure k, n, and p are allowable values, but I am thinking that is all taken care of because all calls will need to go through BinomCoefficientLog eventually.